### PR TITLE
Initial details commit

### PR DESCRIPTION
**Pull request by sanovikov71**
Change-Id: I7c314230e225eeedfdf491c21c804940b9af36eb

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (genaforvena) [inline .idea/compiler.xml]: Please remove idea files! :)

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java [hunk 2]
```diff
@@ -18,10 +21,19 @@
  * Created by imozerov on 05.02.16.
  */
 public class ArtListAdapter extends RecyclerView.Adapter<ArtListAdapter.ViewHolder> {
-    private List<ArtObjectUi> data;
 
-    public ArtListAdapter(@NonNull List<ArtObjectUi> data) {
-        this.data = data;
+    private final Context mContext;
+    private List<ArtObjectUi> mData;
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java:26]: I suggest to not use m, s, a prefixes.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java [hunk 3]
```diff
@@ -29,15 +41,18 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         View rootView = LayoutInflater.from(parent.getContext())
                 .inflate(R.layout.art_object_in_list, parent, false);
         rootView.setOnClickListener((v) -> {
-
+            if (mDetailOpener != null) {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java:44]: I suppose this check is extra

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java [hunk 4]
```diff
@@ -18,10 +21,19 @@
...
+    private List<ArtObjectUi> mData;
+
+    private DetailOpener mDetailOpener;
+
+    public ArtListAdapter(Context context, @NonNull List<ArtObjectUi> data) {
+        mContext = context;
+        this.mData = data;
+
+        if (context instanceof DetailOpener) {
+            mDetailOpener = (DetailOpener) context;
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java:35]: I think we better use 

```
try {
classCast
} catch () {
 throw new RuntimeException("should implemente lalalal")
}
```

as this interface is critical for the adapter

File: app/src/main/java/org/imozerov/streetartview/ui/DetailOpener.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package org.imozerov.streetartview.ui;
+
+/**
+ * Created by sergei on 08.02.16.
+ */
+public interface DetailOpener {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/DetailOpener.java:6]: I think we better name this interface ArtObjectDetailOpener.

File: app/src/main/java/org/imozerov/streetartview/ui/DetailActivity.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+
+import org.imozerov.streetartview.R;
+
+/**
+ * Created by sergei on 08.02.16.
+ */
+public class DetailActivity extends AppCompatActivity {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/DetailActivity.java:12]: renamme to DetailArtObjectActivity

File: app/src/main/java/org/imozerov/streetartview/ui/DetailOpener.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+    void openDetailActivity();
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/DetailOpener.java:8]: We have to pass ArtObject id to this method to be able to distinguish what object we need to show.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java [hunk 8]
```diff
@@ -18,10 +21,19 @@
...
+    private final Context mContext;
+    private List<ArtObjectUi> mData;
+
+    private DetailOpener mDetailOpener;
+
+    public ArtListAdapter(Context context, @NonNull List<ArtObjectUi> data) {
+        mContext = context;
+        this.mData = data;
+
+        if (context instanceof DetailOpener) {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java:34]: Is it really the best way? I have some concerns on re-use of this adapter anywhere else. 
(This comment is not about that this approach is definitely not work. I'm just raising my concerns.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java (hunk 2 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java:26]: ok

**Pull request closed without merging**